imap/cyr_expire.c

```diff
@@ -273,6 +287,44 @@ static int parse_duration(const char *s, int *secondsp)
     return 1;
 }
 
+/*
+ * Given an annotation, reads it, and converts it into 'seconds',
+ * using `parse_duration`.
+ */
+static int get_annotation_value(const char *mboxname,
+                                const char *annot_entry,
+                                int *secondsp, bool iterate)
+{
+    struct buf attrib = BUF_INITIALIZER;
+    int ret;
+    /* mboxname needs to be copied since `mboxname_make_parent`
+     *  runs a strrchr() on it.
+     */
+    char *buf = xstrdup(mboxname);
+
+    /*
+     * Mailboxes inherit /vendo/cmu/cyrus-imapd/{expire, archive, delete},
+     * so we need to iterate all the way up to "" (server entry).
+     */
+    do {
+        buf_free(&attrib);
+        ret = annotatemore_lookup(buf, annot_entry, "", &attrib);
+        if (ret ||              /* error */
+            attrib.s)           /* found an entry */
+            break;
+    } while (mboxname_make_parent(buf) && iterate);
+
+    if (attrib.s && parse_duration(attrib.s, secondsp))
+        ret = 0;
```

Comment:
the truthiness of this function seems to be the inverse of `parse_duration()`?  `parse_duration()` returns 1 on success, 0 on failure, but this returns 0 on success, 1 on failure?